var/spack/repos/builtin/packages/flang/package.py

```diff
@@ -0,0 +1,54 @@
+##############################################################################
+# Copyright (c) 2017, Los Alamos National Security, LLC
+# Produced at the Los Alamos National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+import os
+
+
+class Flang(CMakePackage):
+    """Flang is a Fortran compiler targeting LLVM."""
+    homepage = "https://github.com/flang-compiler/flang"
+    url      = "https://github.com/flang-compiler/flang/flecsi/tarball/v1.0"
+
+    version('develop', git='https://github.com/flang-compiler/flang', branch='master')
+
+    depends_on("llvm@4.0.1+flang")
+
+    def patch(self):
+        # Don't use -Werror
+        # https://github.com/flang-compiler/flang/pull/85
+        filter_file(r'-Werror', '', 'CMakeLists.txt')
+
+    def cmake_args(self):
+        options = [
+            '-DCMAKE_C_COMPILER=%s' % os.path.join(
```

Comment:
looks like it not only depends on `llvm` but needs to be compiled with it. Maybe add `conflicts('%intel`)`, `conflicts('%gcc)` to make sure no-one tries to build it with those.

Reply:
I built `llvm+flang` with `gcc`, it is just that `flang` itself needs to be build with the tools from `llvm+flang`.

Reply:
ah, i see. Ok then.